lib/astarte_core/triggers/policy/policy.ex

```diff
@@ -0,0 +1,180 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2021 Ispirata Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.Core.Triggers.Policy do
+  use Astarte.Core.Triggers.PolicyProtobuf
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Astarte.Core.Triggers.Policy
+  alias Astarte.Core.Triggers.Policy.Handler
+  alias Astarte.Core.Triggers.Policy.TaggedPolicy
+
+  @required_fields [
+    :name,
+    :error_handlers,
+    :maximum_capacity
+  ]
+
+  @permitted_fields [
+    :name,
+    :maximum_capacity,
+    :retry_times,
+    :event_ttl
+  ]
+
+  @primary_key false
+  embedded_schema do
+    field :name
+    field :maximum_capacity, :integer
+    field :retry_times, :integer
+    field :event_ttl, :integer
+    embeds_many :error_handlers, Policy.Handler
+  end
+
+  def changeset(%Policy{} = policy, params \\ %{}) do
+    # TODO check minimum and maxiumim range of attributes + conditions on name
+    policy
+    |> cast(params, @permitted_fields)
+    |> validate_required(@required_fields)
+    |> validate_length(:name, max: 128)
+    |> validate_format(:name, policy_name_regex())
+    |> validate_exclusion(:name, ["astarte", "control", "device"])
+    |> validate_inclusion(:retry_times, 1..100)
+    |> validate_inclusion(:event_ttl, 1..100)
```

Comment:
100 seems quite low for a TTL, maybe I'm missing the unit of measurement

Reply:
We use seconds, however those ranges are somewhat arbitrary right now and could be reconsidered.

Reply:
Imho it should at least allow one day in seconds (i.e. 86400)